--- magento-1.4.1.1/./app/Mage.php~	2010-08-30 21:45:00.000000000 +0300
+++ magento-1.4.1.1/./app/Mage.php	2010-08-31 08:48:08.459842890 +0300
@@ -641,7 +641,7 @@
 
             self::$_isInstalled = false;
 
-            if (is_readable($localConfigFile)) {
+            if (is_readable($localConfigFile) && filesize($localConfigFile)) {
                 $localConfig = simplexml_load_file($localConfigFile);
                 date_default_timezone_set('UTC');
                 if (($date = $localConfig->global->install->date) && strtotime($date)) {
